test/spec/modules/sspBCAdapter_spec.js

```diff
@@ -0,0 +1,366 @@
+
+    it('should return true when required params found', function () {
+      assert(spec.isBidRequestValid(bid));
+    });
+
+    it('should return false when required params are missing', function () {
+      bid.params.id = undefined;
+      assert.isFalse(spec.isBidRequestValid(bid));
+    });
+  });
+
+  describe('buildRequests', function () {
+    const { bids, bidRequest, bidRequestSingle } = prepareTestData();
+    const request = spec.buildRequests(bids, bidRequest);
+    const requestSingle = spec.buildRequests([bids[0]], bidRequestSingle);
+    const payload = request ? JSON.parse(request.data) : { site: false, imp: false };
+    const payloadSingle = request ? JSON.parse(requestSingle.data) : { site: false, imp: false };
+
+    it('should send bid request to endpoint via POST', function () {
+      expect(request.url).to.contain(BIDDER_URL);
+      expect(request.method).to.equal('POST');
+    });
+
+    it('should contain prebid and bidder versions', function () {
+      expect(request.url).to.contain('bdver');
+      expect(request.url).to.contain('pbver=$prebid.version$');
+    });
+
+    it('should create one imp object per bid', function () {
+      expect(payload.imp.length).to.equal(bids.length);
+      expect(payloadSingle.imp.length).to.equal(1);
+    });
+
+    it('should save bidder request data', function () {
+      expect(request.bidderRequest).to.deep.equal(bidRequest);
+    });
+
+    it('should send site Id from bidder params', function () {
+      expect(payload.site.id).to.equal(bids[0].params.siteId);
+    });
+
+    it('should send page url from refererInfo', function () {
+      expect(payload.site.page).to.equal(bidRequest.refererInfo.referer);
+    });
+
+    it('should send gdpr data', function () {
+      expect(payload.regs).to.be.an('object').and.to.have.property('[ortb_extensions.gdpr]', 1);
+      expect(payload.user).to.be.an('object').and.to.have.property('[ortb_extensions.consent]', bidRequest.gdprConsent.consentString);
+    });
+  });
+
+  describe('Ajax tests', function () {
+    const { bids_test, bidRequestTest, bidRequestTestNoGDPR } = prepareTestData();
+    const requestSingleTest = spec.buildRequests(bids_test, bidRequestTest);
+    const requestSingleTestNoGDPR = spec.buildRequests(bids_test, bidRequestTestNoGDPR);
+
+    it('should return test ad when request is made with test:1 and gdpr data', function (done) {
+      const xhr = new XMLHttpRequest();
+      xhr.open(requestSingleTest.method, requestSingleTest.url, true)
+      xhr.onreadystatechange = function() {
+        if (xhr.readyState === 4) {
+          const response = xhr.responseText ? JSON.parse(xhr.responseText) : {};
+          expect(response.seatbid).to.be.an('array');
+          done();
+        }
+      }
+      xhr.send(requestSingleTest.data);
+    });
+
+    it('should return no ads for request without gdpr, regardless of test settings', function (done) {
+      const xhr = new XMLHttpRequest();
```

Comment:
See above

Reply:
@mike-chowla I've removed ajax tests. I have left bidRequestTest and bidRequestTestNoGDPR, which should provide configs that 
- always receive one test ad (bidRequestTest)
- never receive test ad, due to the lack of GDDPR (bidRequestTestNoGDPR)

Could You use config provided in bidRequestTest and check if you're getting a bid response?